src/engine.rs

```diff
@@ -463,15 +464,19 @@ impl Engine {
             .map(|result| Address(result.as_slice().try_into().unwrap()))
     }
 
-    pub fn register_token(&mut self, erc20_token: &[u8], nep141_token: &[u8]) {
+    pub fn register_token(&mut self, erc20_token: &[u8], nep141_account_id: &str) {
+        // Convert nep141 AccountID to lower case
+        let nep141_account_id = nep141_account_id.to_lowercase();
+        let nep141_key = nep141_account_id.as_bytes();
```

Comment:
Some suggestion, instead of doing lower case inside the contract, we can verify if the input is a valid near account id ([the code already exists here](https://github.com/aurora-is-near/aurora-engine/blob/50bf61cf7036e20675010c7399428de08f2e92ba/src/prelude.rs#L65). And fail if it is not a valid account id.

This way it is also clear that we are disallowing invalid account id, instead of trying to convert it to one.

Reply:
I like the suggestion, I mentioned this in #180. The only reason why I don't want to add it here is that I expect that someone may try manually creating the deployment transaction and providing Ethereum checksummed address as an input.

But from the another point of view, it's already NEP-141 input and not ERC-20 and there are no checksum addresses in Near.

Also, I still think that it's ok to have `.to_lowercase` in `get_erc20_from_nep141` (`engine.rs` function) but have a check in `get_erc20_from_nep141` (`lib.rs` function), because otherwise the function in Engine will need to have something like `Result<Option<Vec<u8>>>` as a return type or `Result<Vec<u8>, GetErc20FromNep141Error>`.

@birchmd @joshuajbouw gents, what do you think?

Reply:
Tbh, I think making this lower case looks very arbitrary given that we are only bridging NEP141 accounts which should be valid NEAR accounts. 

I wouldn't do `.to_lowercase` on `get_` either, for consistency + education. Also it is easy to reason about our code. 
I like the approach of compilers, they try to be smart reporting errors, but not too smart fixing them, to avoid unintended consequences or assumptions.

I'm ok in the engine function either returning `Result<Vec<u8>, GetErc20FromNep141Error>` (👈 preferred)  or `Option<Vec<u8>>`. Also we should consider add some logging in case of failure.

Reply:
What I don't like is that this is making a copy of the `nep141_account_id` `&str`. That and we are making redundant copies of `erc20_token` too if I'm not mistaken. This method should take ownership of both parameters and looking at its use, it should be ok.

Reply:
If I recall... And this is just based on previous knowledge. The `to_lowercase` method is a bit "expensive".

The return type you mentioned is fine.